src/facade/dragonfly_connection.cc

```diff
@@ -1157,11 +1163,19 @@ void Connection::Migrate(util::fb2::ProactorBase* dest) {
   // connections
   CHECK(!cc_->async_dispatch);
   CHECK_EQ(cc_->subscriptions, 0);    // are bound to thread local caches
+  CHECK_EQ(self_.use_count(), 1u);    // references cache our thread and backpressure
   CHECK(!dispatch_fb_.IsJoinable());  // can't move once it started
 
   listener()->Migrate(this, dest);
 }
 
+Connection::WeakRef Connection::Borrow() {
+  DCHECK(self_);
+  DCHECK_GT(cc_->subscriptions, 0);
+
+  return WeakRef(self_, queue_backpressure_, socket_->proactor()->GetPoolIndex());
```

Comment:
unrelated but I just confirmed we must fix ` queue_backpressure_ = &tl_queue_backpressure_;` after the migration

Reply:
Yes, I remember this